next-cmake/CMakeLists.txt

```diff
@@ -131,14 +145,30 @@ if(HIPBLASLT_ENABLE_DEVICE)
         # setup. In the meantime, the directory passed here must match the
         # what is set there. Failure to do so will cause a ModuleNotFoundError
         # trying to load rocisa.
-        hipblaslt_configure_bundled_python_command("${CMAKE_BINARY_DIR}/lib")
+        hipblaslt_configure_bundled_python_command("${CMAKE_BINARY_DIR}/lib" "${asan_opts}")
     else()
         # Just pass through to the found python executable and trust that the
         # user set it up properly.
         set(HIPBLASLT_PYTHON_COMMAND "${Python3_EXECUTABLE}")
         set(HIPBLASLT_PYTHON_DEPS)
     endif()
 endif()
+
+if(HIPBLASLT_ENABLE_ASAN)
+    add_library(hipblaslt-asan INTERFACE)
```

Comment:
Clever, but I'm not sure this will work right as this will leak a lot of opinionated things to users of the shared library. If XYZ target gets this as transitive usage requirements, all of the sudden, it is inheriting sanitizer build options and an override of the linker.

These are options you want to build libhipblaslt.so with, but you don't want users of it to inherit them: they have to make their own choices on those things, and there are cases where users will choose differently.

Asan is one of those annoying things that often doesn't fit right in this kind of build... Which is why you see so much ad-hoc helpers for it. I think what you actually want is not easily done with normal cmake target inheritance.